app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -0,0 +1,96 @@
+package org.literacyapp.authentication;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.literacyapp.LiteracyApplication;
+import org.literacyapp.dao.DaoSession;
+import org.literacyapp.model.StudentImage;
+import org.literacyapp.dao.StudentImageDao;
+import org.literacyapp.model.StudentImageFeature;
+import org.literacyapp.dao.StudentImageFeatureDao;
+import org.literacyapp.util.AiHelper;
+import org.opencv.android.OpenCVLoader;
+import org.opencv.core.Mat;
+import org.opencv.core.Rect;
+import org.opencv.core.Size;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.Imgproc;
+
+import java.io.File;
+import java.util.Calendar;
+import java.util.List;
+
+import ch.zhaw.facerecognitionlibrary.PreProcessor.StandardPostprocessing.Resize;
+import ch.zhaw.facerecognitionlibrary.Recognition.Recognition;
+import ch.zhaw.facerecognitionlibrary.Recognition.SupportVectorMachine;
+import ch.zhaw.facerecognitionlibrary.Recognition.TensorFlow;
+
+/**
+ * Created by sladomic on 26.11.16.
+ */
+
+public class TrainingHelper {
+    private Context context;
+    private DaoSession daoSession;
+    private StudentImageDao studentImageDao;
+    private StudentImageFeatureDao studentImageFeatureDao;
+    private SupportVectorMachine svm;
+
+    static {
+        if (!OpenCVLoader.initDebug()) {
+            // Handle initialization error
+        }
+    }
+
+    public TrainingHelper(Context context){
+        this.context = context;
+        LiteracyApplication literacyApplication = (LiteracyApplication) context.getApplicationContext();
+        daoSession = literacyApplication.getDaoSession();
+        studentImageDao = daoSession.getStudentImageDao();
+        studentImageFeatureDao = daoSession.getStudentImageFeatureDao();
+        svm = new SupportVectorMachine(context, Recognition.TRAINING);
+    }
+
+
+    public void extractFeatures(){
+        // Get all the StudentImages where the features haven't been extracted yet
+        List<StudentImage> studentImageList = studentImageDao.queryBuilder()
+                .where(StudentImageDao.Properties.StudentImageFeatureId.eq(0))
```

Comment:
This shouldn't occur (it never occured in the face recogntion app either) but we will need to develop some housekeeping jobs anyway. I think this could be one of them. I opened the issue #226